src/Settings.js

```diff
@@ -82,57 +83,124 @@ const ClearStorage = inject("store")(
+                    <h1>Settings</h1>
+                    <div>
+                        <h2>Morse tone</h2>
+                        <div>
+                            <Slider
+                                id="volume"
+                                label="Voume"
+                                editable
+                                max={100}
+                                min={0}
+                                value={this.props.store.morse.volume}
+                                onChange={value => this.props.store.morse.setVolume(value)}
+                                leftIcon={<FontIcon>build</FontIcon>}
+                            />
+                            <Slider
+                                id="speed"
+                                label="Speed (WPM)"
+                                editable
+                                max={80}
+                                min={10}
+                                value={this.props.store.morse.speed}
+                                onChange={value => this.props.store.morse.setSpeed(value)}
+                                leftIcon={<FontIcon>fast_forward</FontIcon>}
+                            />
+                            <Slider
+                                id="frequency"
+                                label="Frequency (Hz)"
+                                editable
+                                max={1000}
+                                min={200}
+                                step={10}
+                                value={this.props.store.morse.frequency}
+                                onChange={value => this.props.store.morse.setFrequency(value)}
+                                leftIcon={<FontIcon>audiotrack</FontIcon>}
+                            />
+                            <Slider
+                                id="delay"
+                                label="Delay Before Repeat (ms)"
+                                editable
+                                max={5000}
+                                min={10}
+                                step={10}
+                                value={this.props.store.morse.delay}
+                                onChange={value => this.props.store.morse.setDelay(value)}
+                                leftIcon={<FontIcon>build</FontIcon>}
+                            />
+                            <Slider
+                                id="max repeats"
+                                label="Max Repeats"
+                                editable
+                                max={20}
+                                min={1}
+                                step={1}
+                                value={this.props.store.morse.maxRepeats}
+                                onChange={value => this.props.store.morse.setMaxRepeats(value)}
+                                leftIcon={<FontIcon>build</FontIcon>}
+                            />
+                            <Button
```

Comment:
Since the new state added (playCount and playInterval) only has to do with this button, what do you think about turning encapsulating the button and the state into a separate component and leaving the rest of the Settings as it was?